charts/tfy-karpenter-config/files/al2-soci-gpu-provisioner-userdata.sh

```diff
@@ -1,130 +1,57 @@
-SOCI_RELEASE_VERSION="0.7.0"
-SOCI_TAR_CHECKSUM="8766cdd479272dcc86299e70a0f7a9343f940c98285c1491bb3c3cdc05b26f47"
-SOCI_TAR_FILENAME="soci-snapshotter-$SOCI_RELEASE_VERSION-linux-amd64.tar.gz"
-SOCI_CONFIG_DIR="/etc/soci-snapshotter-grpc/"
-SOCI_CONFIG_FILEPATH="$SOCI_CONFIG_DIR/config.toml"
-SOCI_SYSTEMD_SERVICE_FILEPATH="/etc/systemd/system/soci-snapshotter.service"
-cp $CONTAINERD_CONFIG_FILEPATH $BACKUP_CONTAINERD_CONFIG_FILEPATH
-cp $KUBELET_CONFIG_FILEPATH $BACKUP_KUBELET_CONFIG_FILEPATH
-
-setup_soci() {( set -ex
-
-yum install fuse -y
-modprobe fuse
-
-curl \
---silent \
---show-error \
---retry 3 \
---retry-delay 1 \
--L https://github.com/awslabs/soci-snapshotter/releases/download/v${SOCI_RELEASE_VERSION}/soci-snapshotter-${SOCI_RELEASE_VERSION}-linux-amd64.tar.gz \
--o $SOCI_TAR_FILENAME
-echo "$SOCI_TAR_CHECKSUM $SOCI_TAR_FILENAME" | sha256sum -c
-
-tar -C /usr/local/bin -xvf $SOCI_TAR_FILENAME soci soci-snapshotter-grpc
-rm $SOCI_TAR_FILENAME
-
-cat > $SOCI_SYSTEMD_SERVICE_FILEPATH << EOF
-[Unit]
-Description=soci snapshotter containerd plugin
-Documentation=https://github.com/awslabs/soci-snapshotter
-After=network.target
-Before=containerd.service
-
-[Service]
-Type=notify
-ExecStart=/usr/local/bin/soci-snapshotter-grpc
-Restart=always
-RestartSec=5
-
-[Install]
-WantedBy=multi-user.target
-EOF
-
-mkdir -p $SOCI_CONFIG_DIR
 
-cat > $SOCI_CONFIG_FILEPATH << EOF
+# --- Install soci-snapshotter ---
+pushd /tmp
+curl -sSL -o "$ARCHIVE" "https://github.com/awslabs/soci-snapshotter/releases/download/v$VERSION/$ARCHIVE"
+curl -sSL -o "$ARCHIVE.sha256sum" "https://github.com/awslabs/soci-snapshotter/releases/download/v$VERSION/$ARCHIVE.sha256sum"
+sha256sum -c "$ARCHIVE.sha256sum"
+tar -xvzf "$ARCHIVE" -C /usr/local/bin soci-snapshotter-grpc
+rm -f "$ARCHIVE" "$ARCHIVE.sha256sum"
+popd
+
+# --- Configure soci-snapshotter ---
+mkdir -p /etc/soci-snapshotter-grpc
+cat <<EOF > /etc/soci-snapshotter-grpc/config.toml
 [cri_keychain]
-enable_keychain=true
-image_service_path="/run/containerd/containerd.sock"
+enable_keychain = true
+image_service_path = "/run/containerd/containerd.sock"
 EOF
 
+# --- Enable and start soci-snapshotter ---
+curl -sSL -o /etc/systemd/system/soci-snapshotter.service "https://raw.githubusercontent.com/awslabs/soci-snapshotter/v$VERSION/soci-snapshotter.service"
+systemctl daemon-reexec
 systemctl daemon-reload
 systemctl enable --now soci-snapshotter
-systemctl status soci-snapshotter
-
-echo patching $CONTAINERD_CONFIG_FILEPATH
-cat > $CONTAINERD_CONFIG_FILEPATH << EOF
-version = 2
-root = "/var/lib/containerd"
-state = "/run/containerd"
-imports = ["/etc/containerd/config.d/*.toml"]
-
-[grpc]
-address = "/run/containerd/containerd.sock"
 
+# --- Configure containerd to use soci snapshotter ---
+cat > /etc/containerd/config.d/soci.toml <<EOF
 [plugins."io.containerd.grpc.v1.cri".containerd]
-default_runtime_name = "runc"
+default_runtime_name = "nvidia"
```

Comment:
Do we have to set this necessarily? or does the system take care of merging values under `[plugins."io.containerd.grpc.v1.cri".containerd]` from different files?

Reply:
It will merge and if incorrect set will fallback to default runtime `runc`